### PR TITLE
Support XCH fee

### PR DESCRIPTION
**Pull request by kimsk**
This is to resolve the issue https://github.com/richardkiss/chiaswap/issues/3 regarding XCH fee support.

Basically, I add another `input` to capture the fee amount and pass them around accordingly. I have tested with `clean`, `clawback`, and `sweep` cases.

### CODE REVIEW

File: SConstruct [hunk 1]
```diff
@@ -19,7 +19,7 @@ env = Environment(
 
 # Only *.py is included automatically by setup2toml.
 # Add extra 'purelib' files or package_data here.
-py_source = Glob("chiaswap/*.py") + Glob("chiaswap/*.cl")
+py_source = Glob("chiaswap/*.py") + Glob("chiaswap/*.clsp")
```
Comment (kimsk) [inline SConstruct:22]: I think `clsp` is the official `Chialisp` extension, right?

File: chiaswap/main.py [hunk 2]
```diff
@@ -160,13 +160,16 @@ def ui_get_amounts(input, prices):
     )
     print()
     xch_amount = Decimal(input("How much XCH is being traded? > "))
+    print("How much XCH fee (default: 0.00005 mojos)?")
+    print("(You should agree your fee with your counter-party. The fee will be used for either clawback, clean, or sweep spend.)")
+    fee_amount = Decimal(input("> ") or "0.00005")
```
Comment (kimsk) [inline chiaswap/main.py:165]: Adding another input asking for fee amount.

File: chiaswap/main.py [hunk 3]
```diff
@@ -471,8 +475,9 @@ def have_xch_want_btc(logfile, secret_key, btc_amount, xch_amount_mojos):
     puzzle_hash = puzzle_reveal.get_tree_hash()
     address = encode_puzzle_hash(puzzle_hash, "xch")
     xch_amount = Decimal(xch_amount_mojos) / Decimal(int(1e12))
+    fee_amount = Decimal(fee_amount_mojos) / Decimal(int(1e12))
 
-    print(f"go into your XCH wallet and send {xch_amount} XCH to")
+    print(f"go into your XCH wallet and send {xch_amount + fee_amount} XCH to")
```
Comment (kimsk) [inline chiaswap/main.py:480]: The XCH side has to send `xch_amount + fee_amount` to create the coin with the `p2_delayed_or_preimage` puzzle.

File: chiaswap/main.py [hunk 4]
```diff
@@ -481,13 +486,14 @@ def have_xch_want_btc(logfile, secret_key, btc_amount, xch_amount_mojos):
     print()
     clawback_puzzle_hash = ui_get_puzzle_hash(logfile, "enter XCH refund address > ")
 
-    coins = wait_for_payments_to_address(address, xch_amount_mojos)
+    coins = wait_for_payments_to_address(address, xch_amount_mojos + fee_amount_mojos)
     parent_coin_id = fromhex(coins[0]["coin_parent"])
     conditions = [[51, clawback_puzzle_hash, xch_amount_mojos]]
```
Comment (kimsk) [inline chiaswap/main.py:491]: `CREATE_COIN` with `xch_amount_mojos` (not including fee amount).

File: chiaswap/main.py [hunk 5]
```diff
@@ -619,7 +626,7 @@ def have_btc_want_xch(logfile, secret_key, btc_amount, xch_amount_mojos):
 
     while True:
         print(
-            "Enter your counterparty private key OR the lightning invoice receipt pre-image or `quit`"
+            "Enter your counter-party private key OR the lightning invoice receipt pre-image or `quit`"
```
Comment (richardkiss) [inline chiaswap/main.py]: A cursory Google search suggests that "counterparty" is a word and that it makes sense here.

File: chiaswap/main.py [hunk 6]
```diff
@@ -715,7 +725,7 @@ def handle_sweep_preimage(
     spend_bundle_hex = bytes(spend_bundle).hex()
     print(f"sweep spend bundle: {spend_bundle_hex}")
     print()
-    print("Your counterparty should share their (disposable) private key")
+    print("Your counter-party should share their (disposable) private key")
```
Comment (richardkiss) [inline chiaswap/main.py]: Ditto.

**Review by richardkiss — CHANGES_REQUESTED** [review]
Undo the "counterparty" changes and I'll merge. Sound good?

Sorry for the delay.

**Comment by kimsk** [comment]
> Undo the "counterparty" changes and I'll merge. Sound good?

Undo, and I also notice coupIe more places using `counter-party`, so I changed them all to counterparty.

**Comment by richardkiss** [comment]
> > Undo the "counterparty" changes and I'll merge. Sound good?
> 
> Undo, and I also notice coupIe more places using `counter-party`, so I changed them all to counterparty.

Ha, thanks for this consistency improvement.

**Pull request merged**